components/Dashboard/Settings/SocialFormField.tsx

```diff
@@ -0,0 +1,107 @@
+import { toUpper } from 'lodash'
+import React from 'react'
+import { FieldError, ValidationOptions } from 'react-hook-form'
+import { useTranslation } from '../../../config/i18n'
+import FacebookIcon from '../../Icons/FacebookIcon'
+import GlobeIcon from '../../Icons/GlobeIcon'
+import InstagramIcon from '../../Icons/InstagramIcon'
+import XIcon from '../../Icons/XIcon'
+import YoutubeIcon from '../../Icons/YoutubeIcon'
+
+export const SocialPlatform = {
+  FACEBOOK: 'FACEBOOK',
+  YOUTUBE: 'YOUTUBE',
+  INSTAGRAM: 'INSTAGRAM',
+  WEBSITE: 'WEBSITE',
+}
+
+const socialMediaPatterns = [
+  {
+    platform: SocialPlatform.FACEBOOK,
+    pattern: /^(?:(?:http|https):\/\/)?(?:www\.)?(?:facebook\.com|fb\.com)\/([A-Za-z0-9-_\.]+)\/?$/im,
+  },
+  {
+    platform: SocialPlatform.INSTAGRAM,
+    pattern: /^(?:(?:http|https):\/\/)?(?:www\.)?(?:instagram\.com|instagr\.am)\/([A-Za-z0-9-_\.]+)\/?$/im,
+  },
+  {
+    platform: SocialPlatform.YOUTUBE,
+    pattern: /^(?:(?:http|https):\/\/)?(?:www\.)?(?:youtube\.com)\/(channel|user)\/([A-Za-z0-9-_\.]+)\/?$/im,
+  },
+  {
+    platform: SocialPlatform.WEBSITE,
+    pattern: /^(?:(?:http|https):\/\/)?(?:www\.)?(?:[a-z0-9](?:[a-z0-9-]{0,61}[a-z0-9])?\.)+[a-z0-9][a-z0-9-]{0,61}[a-z0-9]\/?$/im,
+  },
+]
+
+const PlatformIcon: React.FC<{ name: string }> = ({ name }) => {
```

Comment:
I think we can make the type of `name` to be `SocialPlatform` just to get a little extra type safety and not need the `<XIcon />` backup